### PR TITLE
Add documentation for 7005 errors

### PR DESCRIPTION
**Pull request by Morriar**
image

### CODE REVIEW

File: website/docs/error-reference.md [hunk 1]
```diff
@@ -1047,6 +1047,59 @@ bar(1) # error
...
+sig { params(name: String).returns(Configuration) }
+def find(name)
+  config = @lookup[name]
+  raise ArgumentError, "Configuration #{name} not found" unless config
+  config
+end
+```
+
+By using a local variable, we allow Sorbet to assert that `config` is never
+nilable passed the `raise` instruction.
```
Comment (aprocter-stripe) [inline website/docs/error-reference.md:1101]:
```suggestion
nilable past the `raise` instruction.
```

File: website/docs/error-reference.md [hunk 2]
```diff
@@ -1047,6 +1047,59 @@ bar(1) # error
...
+is to use `T.must` when returning the value:
+
+```rb
+sig { params(name: String).returns(Configuration) }
+def find(name)
+  T.must(@lookup[name])
+end
+```
+
+In some case, we're already being cautious and perform some checks before
```
Comment (aprocter-stripe) [inline website/docs/error-reference.md:1075]:
```suggestion
In some cases, we're already being cautious and perform some checks before
```

File: website/docs/error-reference.md [hunk 3]
```diff
@@ -1047,6 +1047,59 @@ bar(1) # error
 bar(x: 1) # ok
 ```
 
+## 7005
+
+Sorbet detected a mismatch between the declared return type for the method and
+the type of the returned value.
+
```
Comment (aprocter-stripe) [inline website/docs/error-reference.md:1054]: (non-blocking suggestion) Might be good to throw in an ultra-obvious example to start with, something along the lines of:

```rb
sig { returns(Integer) }
def answer
  "42" # error: Expected `Integer` but found `String("42")` for method result type 
end
```

File: website/docs/error-reference.md [hunk 4]
```diff
@@ -1047,6 +1047,59 @@ bar(1) # error
...
+```rb
+sig { params(name: String).returns(Configuration) }
+def find(name)
+  raise ArgumentError, "Configuration #{name} not found" unless @lookup.key?(name)
+  @lookup[name] # error: Expected `Configuration` but found `T.nilable(Configuration)` for method result type
+end
+```
+
+While this code is correct, Sorbet cannot assert the state of `@lookup` didn’t
```
Comment (aprocter-stripe) [inline website/docs/error-reference.md:1086]: Maybe `assume` or `infer` instead of `assert`?

**Review by aprocter-stripe — CHANGES_REQUESTED** [review]
Thanks much for this! Just a couple of small typos/wording suggestions, and one non-blocking suggestion for an additional example.

File: website/docs/error-reference.md (hunk 3 above)
Comment (Morriar) [reply website/docs/error-reference.md:1061]: Good idea 👍

**Review by aprocter-stripe — APPROVED** [review]
Thanks!

**Comment by aprocter-stripe** [comment]
Sorry for the delayed merge here! :)

**Pull request merged**